src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php

```diff
@@ -90,6 +92,27 @@ public function supportsNormalization($data, string $format = null)
      * @throws NotNormalizableValueException
      */
     public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        try {
+            $result = $this->doDenormalize($data, $type);
+        } catch (NotNormalizableValueException $exception) {
+            if ($context[self::COLLECT_INVARIANT_VIOLATIONS] ?? false) {
+                $violation = new InvariantViolation($data, 'This value is not a valid data URI.', $exception);
```

Comment:
Maybe we could have a more explicit or accurate message by reusing the one from the exception ?
I won't repeat for every instances ;)

Reply:
I made the code use the same messages as exceptions for all failure results.

I'd like the `DenormalizationResult` to integrate with `symfony/translation` somehow, e.g. like `symfony/validator` does, but that may be done in a later iteration.